pkg/piecestore/psserver/psdb/psdb.go

```diff
@@ -245,15 +258,45 @@ func (db *DB) WriteBandwidthAllocToDB(rba *pb.Order) error {
 	// If the agreements are sorted we can send them in bulk streams to the satellite
 	t := time.Now()
 	startofthedayunixsec := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location()).Unix()
-	_, err = db.DB.Exec(`INSERT INTO bandwidth_agreements (satellite, agreement, signature, uplink, serial_num, total, max_size, created_utc_sec, expiration_utc_sec, action, daystart_utc_sec) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
+	_, err = db.DB.Exec(`INSERT INTO bandwidth_agreements (satellite, agreement, signature, uplink, serial_num, total, max_size, created_utc_sec, status, expiration_utc_sec, action, daystart_utc_sec) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
 		rba.PayerAllocation.SatelliteId.Bytes(), rbaBytes, rba.GetSignature(),
 		rba.PayerAllocation.UplinkId.Bytes(), rba.PayerAllocation.SerialNumber,
-		rba.Total, rba.PayerAllocation.MaxSize, rba.PayerAllocation.CreatedUnixSec,
+		rba.Total, rba.PayerAllocation.MaxSize, rba.PayerAllocation.CreatedUnixSec, bwaStatusUNSENT,
 		rba.PayerAllocation.ExpirationUnixSec, rba.PayerAllocation.GetAction().String(),
 		startofthedayunixsec)
 	return err
 }
 
+// DeleteBandwidthAllocationPayouts delete paid and/or old payout enteries based on days old
+func (db *DB) DeleteBandwidthAllocationPayouts() error {
+	defer db.locked()()
+
+	//@TODO make a config value for older days
+	t := time.Now().Add(time.Hour * 24 * -90).Unix()
+	_, err := db.DB.Exec(`DELETE FROM bandwidth_agreements WHERE created_utc_sec < ?`, t)
+	if err == sql.ErrNoRows {
+		err = nil
+	}
+	return err
+}
+
+// UpdateBandwidthAllocationStatus update the bwa payout status
+func (db *DB) UpdateBandwidthAllocationStatus(serialnum string, status pb.AgreementsSummary_Status) (err error) {
+	defer db.locked()()
+
+	switch status {
+	case pb.AgreementsSummary_OK:
+		//set update status to "SENT"
+		_, err = db.DB.Exec(`UPDATE bandwidth_agreements SET status = ? WHERE serial_num = ?`, bwaStatusSENT, serialnum)
+	default:
+		// received REJECT(due to bad signature or any data in rba changed) or
+		// FAIL(due to serialnum duplicates) from satellite, set the bwa status to "REJECT"
+		_, err = db.DB.Exec(`UPDATE bandwidth_agreements SET status = ? WHERE serial_num = ?`, bwaStatusREJECT, serialnum)
```

Comment:
Rejected != Fail
The storage node should retry the failed bwagreements but don't retry the rejected bwagreement.

Reply:
updated ..  when 'fail' received from satellite, bwa status remains "unsent"